### PR TITLE
Btag s fs

### PR DESCRIPTION
**Pull request by valeriadamante**
PR to implement the BTAg working point based (not shape!!) scale factors uncertainties

### CODE REVIEW

File: btag.h [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+    bTagCorrProvider(const std::string& fileName, const std::string& efficiencyFileName) :
+        corrections_(CorrectionSet::from_file(fileName)),
+        deepJet_incl_(corrections_->at("deepJet_incl")),
+        deepJet_comb_(corrections_->at("deepJet_comb")),
+        deepJet_wp_values_(corrections_->at("deepJet_wp_values"))
+    {
+        if(efficiencyFileName.size()>0){
+            auto efficiencyFile = root_ext::OpenRootFile(efficiencyFileName);
+            const std::vector<std::string> WpNames = {"Loose", "Medium", "Tight"};
+            const std::vector<int> Flavours = {0, 4, 5};
```
Comment (kandrosov) [inline btag.h:72]:
```suggestion
            static const std::vector<std::string> WpNames = {"Loose", "Medium", "Tight"};
            static const std::vector<int> Flavours = {0, 4, 5};
```

File: btag.h [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+                    num->Divide(denum);
+                    histMapEfficiency[std::make_pair(getWPFromString(WPName),flav)] = std::shared_ptr<TH2>(num);
+                }
+            }
+        }
+    }
+
+    float getWPvalue(WorkingPointsbTag wp) const {
+        return deepJet_wp_values_->evaluate({getWPStr(wp)});
+    }
```
Comment (kandrosov) [inline btag.h:86]: load wp map in the constructor -> converting to string each time and call evaluate is way to inefficient

File: btag.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+import yaml
+
+
+
+class bTagCorrProducer:
+    jsonPath = "/cvmfs/cms.cern.ch/rsync/cms-nanoAOD/jsonpog-integration/POG/BTV/{}/btagging.json.gz"
+    bTagEff_JsonPath = "Corrections/data/BTV/{}/btagEff_TT.root"
+    initialized = False
+    SFSources = ["btagSFbc", "btagSFlight"]
+    correlation = ["nocorr","correlated", "uncorrelated"]
```
Comment (kandrosov) [inline btag.py:14]: same here - don't introduce correlation, only sources

**Review by kandrosov — CHANGES_REQUESTED** [review]
remove btagEff.root, keep only btagEff_TT.root

File: btag.h [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+    enum class UncSource : int {
+        Central = -1,
+        btagSFbc = 0,
+        btagSFlight = 1,
+
+    };
+    enum class Correlation : int {
+        nocorr = -1,
+        uncorrelated = 0,
+        correlated = 1,
```
[the remark this answers is not in the document]
Comment (valeriadamante) [reply btag.h:18]: The names are: syst_up/down_correlated/uncorrelated

File: btag.h (hunk 1 above)
Comment (valeriadamante) [reply btag.h:72]: ok

File: btag.h (hunk 4 above)
[the remark this answers is not in the document]
Comment (kandrosov) [reply btag.h:18]: if needed you can have Correlation internally, but from outside it should be only UncSource and UncScale, where "correlated" and "uncorrelated" are different sources.

File: btag.h (hunk 2 above)
Comment (valeriadamante) [reply btag.h:86]: ok

File: btag.py (hunk 3 above)
Comment (valeriadamante) [reply btag.py:14]: The names are: source_up/down_correlated/uncorrelated

Comment (kandrosov) [reply btag.py:14]: at the python level only unc source should be used.

**Pull request merged**